frame/contracts/src/exec.rs

```diff
@@ -203,8 +203,8 @@ pub trait Ext: sealing::Sealed {
 		take_old: bool,
 	) -> Result<WriteOutcome, DispatchError>;
 
-	/// Returns a reference to the account id of the caller.
-	fn caller(&self) -> &AccountIdOf<Self::T>;
+	/// Returns the caller.
+	fn caller(&self) -> Origin<Self::T>;
```

Comment:
@athei the reason is the implementation of that `caller` function
```rust
	fn caller(&self) -> Origin<T> {
		if let Some(caller) = &self.top_frame().delegate_caller {
			caller.clone()
		} else {
			self.frames()
				.nth(1)
                                 // next line returns a reference to data owned by the current function
				.map(|f| Origin::from_account_id(f.account_id.clone()))
				.unwrap_or(self.origin.clone())
		}
	}
```

If we want to return a reference we could refactor the struct `Frame` to hold a `Origin<T>` instead of `T::AccountId`.
Alternatively, I guess we could workaround it with a lifetime parameter and probably a `transmute`.

Reply:
This does not sound like it is worth it. I guess we rather live with the copy here.